File: MathsThing_Plugins/src/uk/ac/uea/mathsthing/plugins/functions/ACos.java
package uk.ac.uea.mathsthing.plugins.functions;

import java.math.BigDecimal;

import uk.ac.uea.mathsthing.IPlugin.IFunctionPlugin;
import uk.ac.uea.mathsthing.util.FormulaException;

public class ACos extends IFunctionPlugin {

	@Override
	public String getName()
	{
		return "acos";
	}

	@Override
	public BigDecimal function(BigDecimal input) throws FormulaException
	{
		return new BigDecimal(Math.acos(input.doubleValue()));
	}

}

File: MathsThing_Plugins/src/uk/ac/uea/mathsthing/plugins/functions/ACos.scala
package uk.ac.uea.mathsthing.plugins.functions

import java.math.BigDecimal

import uk.ac.uea.mathsthing.IPlugin.IFunctionPlugin

class ACos extends IFunctionPlugin {

  def getName : String = "acos"
    
  def function(input: BigDecimal) : BigDecimal = {
    if (input.doubleValue == Double.NaN 
  	    || input.doubleValue <= Double.NegativeInfinity 
  	    || input.doubleValue >= Double.PositiveInfinity)
      null
    
    try {
      val result = new BigDecimal(Math.acos(input.doubleValue))
      
      if (result.doubleValue == Double.NaN 
  	      || result.doubleValue <= Double.NegativeInfinity 
  	      || result.doubleValue >= Double.PositiveInfinity 
  	      || result.doubleValue > 1000 || result.doubleValue < -1000)
        null;
  	  else
        result
    } catch {
      case ex:NumberFormatException => null
    }
  }
}